### ISSUE TITLE
Inflow at an angle with BDS

### ISSUE THREAD

**Issue by marchdf** [AMReX-Codes/AMReX-Hydro#95]
Hi all,

I am observing the following in amr-wind with BDS and @esclapez is seeing the same in Pele, which is why I think this is a problem with the BDS implementation in AMReX-Hydro and not a problem with our codes/BCs.

Description of the problem:
- xlo and ylo: inflow at u = v = 5 m/s, w = 0
- xhi and yhi: outflow
- zhi and zlo: slip wall

We see corner effects that are not there with PPM/PLM. The flow should be constant everywhere but it is not and we see artifacts at the corners:
<img width="1148" alt="Screen Shot 2022-08-30 at 14 40 17" src="https://user-images.githubusercontent.com/15038415/187538891-3b82c5ec-a2f3-4584-b45b-addf3dcfd9fb.png">

Lucas reports that it looks fine in 2D but not 3D in Pele.

**Comment by ajnonaka** [comment]
In IAMR/Exec/run3d I set up this same case (but with vel=(1,1,0)) and tested with this inputs file:
https://github.com/ajnonaka/IAMR/blob/bds_diagonal/Exec/run3d/regtest.3d.traceradvect_bds
You'll see there a Godunov_PPM and BDS option.  The runs both give the same correct result (the velocity gets knocked off (1,1,0) in the 12th digit only in both options).  I believe the BDS is being invoked because the plotfiles do diff differently (10^-27 in velocity, 10^-5 in the tracer).
Is there an easy way to provide a broken setup?

**Comment by marchdf** [comment]
Thanks for helping us out @ajnonaka. I can provide an input file for amr-wind: 
[freestream_godunov.inp.txt](https://github.com/AMReX-Codes/AMReX-Hydro/files/9465275/freestream_godunov.inp.txt)
You can just build amr-wind and run the resulting executable on that.

But given the similarities of IAMR and PeleLM(eX) maybe it's best/easier for you to start there? @esclapez would have an input file for that.

**Comment by ajnonaka** [comment]
Ann and I were unable to reproduce this problem with IAMR.  If somebody has an example they can point to (IAMR, PeleLM, amr-wind, etc.) please re-open the issue and provide details.  @marchdf I was told to hold off on your amr-wind example for now, unless there have been developments since.

**Comment by marchdf** [comment]
Writing this for posterity. 

After much digging around and comparisons between IAMR and amr-wind, I came to the conclusion (later verified through a conversation with @cgilet) that amr-wind was missing an assumption that the BDS code made. Namely, BDS assumes that umac has 1 filled ghost cell. IAMR and incflo both have a version of `create_umac_grown` that takes care of physical BCs. amr-wind wasn't doing this. I tried manually filling the umac ghost with what I expected and the corner issue goes away. My next task is to replace with an actual fillpatch (a-la IAMR/incflo) and I expect things to work after that. 

@cgilet pointed out that missing this functionality for Godunov might be a problem in multilevel simulations but might only be visible in certain cases. TBD...
